tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,455 @@
+
+        The `mock_load_data` fixture should be automatically used if we don't have the real data
+        downloaded.
+        """
+        fake_input_dir: Path = tmp_path_factory.mktemp("profet_data")
+
+        # We expect the `load_data` function to NOT attempt to download the dataset, hence the timeout.
+        fake_x, fake_y, fake_c = self.Task.ModelConfig().load_data(fake_input_dir)
+        # NOTE: This might look a bit weird, but it's consistent for each of the real datasets.
+        assert fake_x.shape[0] == fake_y.shape[1] == fake_c.shape[1]
+
+    @pytest.mark.skipif(
+        not is_nonempty_dir(REAL_PROFET_DATA_DIR), reason="Real profet data is required.",
+    )
+    @pytest.mark.timeout(15)
+    def test_mock_load_data_fixture_when_real_data_available(
+        self, tmp_path_factory, mock_load_data
+    ):
+        """Test that the mock_load_data fixture returns real data when the input dir is non-empty,
+        and that the means, dtypes, shapes, etc match between the real and fake data.
+        """
+        # NOTE: Need to re-import, because the `mock_load_data` fixture modifies that function in the
+        # module. If we imported the function at the top of this test module, we would get the original
+        # version.
+
+        model_config = self.Task.ModelConfig()
+
+        real_input_dir: Path = REAL_PROFET_DATA_DIR
+        real_x, real_y, real_c = model_config.load_data(real_input_dir)
+
+        fake_input_dir: Path = tmp_path_factory.mktemp("profet_data")
+        fake_x, fake_y, fake_c = model_config.load_data(fake_input_dir)
+        assert real_x.shape == fake_x.shape
+        assert real_y.shape == fake_y.shape
+        assert real_c.shape == fake_c.shape
+
+        assert real_x.dtype == fake_x.dtype
+        assert real_y.dtype == fake_y.dtype
+        assert real_c.dtype == fake_c.dtype
+
+        assert 0 <= real_x.min() and 0 <= fake_x.min()
+        assert real_x.max() <= 1 and fake_x.max() <= 1
+
+        min_y, max_y = y_min[model_config.benchmark], y_max[model_config.benchmark]
+        assert min_y <= real_y.min() and min_y <= fake_y.min()
+        assert real_y.max() <= max_y and fake_y.max() <= max_y
+
+        min_c, max_c = c_min[model_config.benchmark], c_max[model_config.benchmark]
+        assert min_c <= real_c.min() and min_c <= fake_c.min()
+        assert real_c.max() <= max_c and fake_c.max() <= max_c
+
+    @pytest.mark.timeout(30)
+    @pytest.mark.parametrize("device_str", _devices)
+    def test_attributes(
+        self,
+        profet_train_config: MetaModelConfig,
+        profet_input_dir: Path,
+        device_str: str,
+        tmp_path_factory,
+    ):
+        """Simple test: Check that a task can be created and that its attributes are used and set
+        correctly.
+        """
+        max_trials = 123
+        checkpoint_dir: Path = tmp_path_factory.mktemp("checkpoints")
```

Comment:
Why not use the fixture checkpoint_dir instead?

Reply:
Because that fixture is session-scoped, and this expects an empty folder. Using the other one has side-effects on the other tests.

Reply:
Would it make sense to make it have a function scope instead or that adds a significant overhead to many tests?

Reply:
It would add some overhead.

Reply:
Sure, but is it significant enough to justify the additional boilerplate?

Reply:
Fixed in 61cf56505616cdee6da4874b04291b495467ad93